pkg/controller/storagecluster/reconcile.go

```diff
@@ -66,20 +146,35 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1alpha1.StorageClust
 	// Check if this CephCluster already exists
 	found := &rookCephv1.CephCluster{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Name: cephCluster.Name, Namespace: cephCluster.Namespace}, found)
-
 	if err != nil && errors.IsNotFound(err) {
 		reqLogger.Info("Creating CephCluster")
-		err = r.client.Create(context.TODO(), cephCluster)
-
-	} else if err == nil {
-		// Update the CephCluster if it is not in the desired state
-		if !reflect.DeepEqual(cephCluster.Spec, found.Spec) {
-			reqLogger.Info("Updating spec for CephCluster")
-			found.Spec = cephCluster.Spec
-			err = r.client.Update(context.TODO(), found)
-		}
+		return r.client.Create(context.TODO(), cephCluster)
+	}
+	if err != nil {
```

Comment:
Nit: I see why you would move the following logic out of its own stanza, but I don't like this solution. Please combine this if statement with the previous one, i.e. something like:
```
if err != nil {
  if NotFound() {
    logInfo()
    err = Create()
  }
  return err
}
```